pex/interpreter_constraints.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
This new library is extent of the intersecting interpreter constraints logic that makes sense to move out of Pants and into Pex. It will be cleanly consumable by Pants and the only modification I made is to add a meet_all_constraints kwarg for meeting all filters passed instead of at least one, which is backward compatible with the Pants use cases. 

I added a unit test for the function I wrote myself, parse_interpreter_constraints. I operated on the assumption that the other methods of this lib have been tested or are functioning properly so I did not add any new tests for them but I am more than happy to. I also realize tests in this changeset (pex_bootstrapper, interpreter_constraints tests) are relatively sparse but I figured I would post for review to get feedback before putting significant time into them.

Reply:
Should this get bumped to 2017?